app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/map/ActivitiesMapScreenTest.kt

```diff
@@ -181,4 +179,124 @@ class ActivitiesMapScreenTest {
+    composeTestRule.onNodeWithTag("ActivityDate").assertExists().assertTextEquals("20/12/2024")
+
+    // Verify time (expected format: "12:00")
+    composeTestRule.onNodeWithTag("ActivityTime").assertExists().assertTextEquals("12:00")
+
+    // Verify location
+    composeTestRule.onNodeWithTag("ActivityLocation").assertExists().assertTextEquals("Paris")
+
+    // Verify description
+    composeTestRule
+        .onNodeWithTag("ActivityDescription")
+        .assertExists()
+        .assertTextEquals("Monthly team meeting to discuss project progress.")
+  }
+
+  @Test
+  fun testGpsDetailsContentDisplaysCorrectDetails() {
+    composeTestRule.setContent {
+      GpsDetailsContent(
+          gpsRouteDetails =
+              RouteDetails(
+                  origin = LatLng(48.8566, 2.3522), // Mock origin (e.g., Paris)
+                  destination = LatLng(49.8566, 2.3522), // Mock destination
+                  route = listOf(LatLng(48.8566, 2.3522), LatLng(49.8566, 2.3522)), // Mock route
+                  legsDistance = listOf("5 km"),
+                  legsDuration = listOf("10 mins"),
+                  legsRoute =
+                      listOf(
+                          listOf(
+                              LatLng(48.8566, 2.3522),
+                              LatLng(48.8570, 2.3530)) // Mock GPS to activity route
+                          )),
+          selectedActivity = listOfActivities[0] // Mock selected activity
+          )
+    }
+
+    // Verify title
+    composeTestRule
+        .onNodeWithTag("RouteDetailsTitle_Gps")
+        .assertExists()
+        .assertTextEquals("Your current Location")
+
+    // Verify distance
+    composeTestRule
+        .onNodeWithTag("RouteDetailsDistance_Gps")
+        .assertExists()
+        .assertTextEquals("Distance: 5 km")
+
+    // Verify duration
+    composeTestRule
+        .onNodeWithTag("RouteDetailsDurationGps")
+        .assertExists()
+        .assertTextEquals("Duration: 10 mins")
+
+    // Verify subtitle (Activity Title)
+    composeTestRule
+        .onNodeWithTag("RouteDetailsSubtitle_Gps")
+        .assertExists()
+        .assertTextEquals("Team Meeting")
+  }
+
+  @Test
+  fun testLegDetailsContentDisplaysCorrectDetails() {
+    composeTestRule.setContent {
+      LegDetailsContent(
+          routeDetails =
+              RouteDetails(
+                  origin = LatLng(48.8566, 2.3522), // Mock origin (Paris)
+                  destination = LatLng(49.8566, 2.3522), // Mock destination
+                  route = listOf(LatLng(48.8566, 2.3522), LatLng(49.8566, 2.3522)), // Mock route
+                  legsDistance = listOf("3.4 km", "5.2 km"),
+                  legsDuration = listOf("15 mins", "25 mins"),
+                  legsRoute =
+                      listOf(
+                          listOf(LatLng(48.8566, 2.3522), LatLng(48.8570, 2.3530)), // Mock leg 1
+                          listOf(LatLng(48.8570, 2.3530), LatLng(49.8566, 2.3522)) // Mock leg 2
```

Comment:
the first listOf is the path between activity 1 and 2 and de second listOf is the path between activity 2 and 3

Reply:
No, since the last time you reviewed the code I changed the implementation. At the beginning I wanted to have the possibility to display different path between activity 1 and 2. but in the meantime I realized that this would be useless so there is only one path between the two.

like I said, in my test : the first listOf is the path (what I call leg) between activity 1 and 2. this path is composed by latLng point

Reply:
Oh, ok. So no matter what the list only contains two elements each time, which are the coordinate of the two endpoints ?

Reply:
No, legsRoute contains the list of all legs for all activities. So if there are 3 activities, there will be 2 legs (path). So legsRoute = list(leg0, leg1). 

Each leg is made up of a list of latLng pointsfor the drawing of the leg. The number of latLng is determined by google and I can be greater than 2. So leg0 = (latlng0, latLng1,....)

Reply:
After a discussion, I now understand how the leg concept works. So a leg is all the points needed to construct a path between two activities. These are put in a list. And then, you have a list of all paths, so a list of lists of coordiantes